lib/routes.py

```diff
@@ -81,8 +108,16 @@ def on_join_room(data):
 @socketio.on('leave')
 def on_leave(data):
   room = data['room']
+  return_group = data['return_to']
   leave_room(room)
   send(f'A User left the {room} room', room=room)
+  if return_group:
+    my_conn = Connection.objects(sid=request.sid).first()
+    my_conn.waiting = True
+    my_conn.save()
+    join_room(return_group)
+    matchmake(return_group)
+
```

Comment:
We can work on the leave section with either front end to add a leave button or we can tackle that timer method next.